Fixes warnings of this form:

../test/testfile.c: In function ‘test_file_readable’:
../test/testfile.c:43:22: warning: ‘~’ on a boolean expression [-Wbool-operation]
   43 |     else if (!RES && ~EXPECTED) \
      |                      ^
../test/testfile.c:53:3: note: in expansion of macro ‘READABLE’
   53 |   READABLE (false, "non-existing-file");
      |   ^~~~~~~~
../test/testfile.c:43:22: note: did you mean to use logical not?
   43 |     else if (!RES && ~EXPECTED) \
      |                      ^
../test/testfile.c:53:3: note: in expansion of macro ‘READABLE’
   53 |   READABLE (false, "non-existing-file");
      |   ^~~~~~~~

--- kissat-rel-4.0.4/test/testfile.c.orig	2025-10-16 11:24:28.000000000 -0600
+++ kissat-rel-4.0.4/test/testfile.c	2025-10-16 13:41:05.112400640 -0600
@@ -40,7 +40,7 @@ static void test_file_readable (void) {
     bool RES = kissat_file_readable (PATH); \
     if (RES && EXPECTED) \
       printf ("file '%s' determined to be readable as expected\n", PATH); \
-    else if (!RES && ~EXPECTED) \
+    else if (!RES && !EXPECTED) \
       printf ("file '%s' determined not to be readable as expected\n", \
               PATH); \
     else if (RES && !EXPECTED) \
@@ -66,7 +66,7 @@ static void test_file_writable (void) {
     bool RES = kissat_file_writable (PATH); \
     if (RES && EXPECTED) \
       printf ("file '%s' determined to be writable as expected\n", PATH); \
-    else if (!RES && ~EXPECTED) \
+    else if (!RES && !EXPECTED) \
       printf ("file '%s' determined not to be writable as expected\n", \
               PATH); \
     else if (RES && !EXPECTED) \
